File: tests/fixtures/dotnet/Controllers/CommentedController.cs
using Microsoft.AspNetCore.Mvc;

namespace TestApi.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class CommentedController : ControllerBase
    {
        // Single line commented endpoints - should be filtered
        // [HttpGet("single-line-comment")]
        // public IActionResult GetSingleLineComment() { return Ok("filtered"); }

        // [HttpPost("another-single-line")]
        // public IActionResult PostAnotherSingleLine() { return Ok("filtered"); }

        /* Block commented endpoints - should be filtered */
        /* [HttpPut("block-comment")] */
        /* public IActionResult PutBlockComment() { return Ok("filtered"); } */

        /*
         * Multi-line block commented endpoints - should be filtered
         * [HttpDelete("multi-line-block")]
         * public IActionResult DeleteMultiLineBlock() { return Ok("filtered"); }
         */

        /// <summary>
        /// XML doc commented endpoints - should be filtered
        /// [HttpPatch("xml-doc-comment")]
        /// public IActionResult PatchXmlDocComment() { return Ok("filtered"); }
        /// </summary>

        // Active endpoints - should NOT be filtered
        [HttpGet("active")]
        public IActionResult GetActive()
        {
            return Ok("active");
        }

        [HttpPost("users")]
        public IActionResult CreateUser()
        {
            return Ok("created");
        }

        // Mixed scenarios
        /*
        [HttpGet("mixed-block")]
        public IActionResult GetMixedBlock()
        {
            return Ok("filtered");
        }
        */

        // [HttpGet("commented-inline")] // This should be filtered

        [HttpPatch("active-after-comment")]
        public IActionResult PatchActiveAfterComment()
        {
            return Ok("active"); // This should NOT be filtered
        }

        // Route attribute variations
        // [Route("commented-route", Name = "CommentedRoute")]
        // [HttpGet]
        // public IActionResult GetCommentedRoute() { return Ok("filtered"); }

        [Route("active-route", Name = "ActiveRoute")]
        [HttpGet]
        public IActionResult GetActiveRoute()
        {
            return Ok("active");
        }
    }
}

File: tests/fixtures/nestjs/src/commented.controller.ts
import { Controller, Get, Post, Put, Delete, Patch } from '@nestjs/common';

@Controller('comments')
export class CommentedController {

  // Single line commented endpoints - should be filtered
  // @Get('/single-line-comment')
  // getSingleLineComment() { return 'filtered'; }

  // @Post('/another-single-line')
  // postAnotherSingleLine() { return 'filtered'; }

  /* Block commented endpoints - should be filtered */
  /* @Put('/block-comment') */
  /* putBlockComment() { return 'filtered'; } */

  /*
   * Multi-line block commented endpoints - should be filtered
   * @Delete('/multi-line-block')
   * deleteMultiLineBlock() { return 'filtered'; }
   */

  /**
   * JSDoc commented endpoints - should be filtered
   * @Patch('/jsdoc-comment')
   * patchJsdocComment() { return 'filtered'; }
   */

  // Active endpoints - should NOT be filtered
  @Get('/active')
  getActive() {
    return 'active';
  }

  @Post('/users')
  createUser() {
    return 'created';
  }

  // Mixed scenarios
  /*
  @Get('/mixed-block')
  getMixedBlock() {
    return 'filtered';
  }
  */

  // @Get('/commented-inline') // This should be filtered

  @Patch('/active-after-comment')
  patchActiveAfterComment() {
    return 'active'; // This should NOT be filtered
  }

  // GraphQL examples
  // @Query(() => String)
  // getCommentedQuery() { return 'filtered'; }

  @Query(() => String)
  getActiveQuery() {
    return 'active';
  }
}